core/src/main/java/com/linecorp/armeria/common/ContextStorage.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.util.UnstableApi;
+
+/**
+ * Storage.
+ */
+@UnstableApi
+public interface ContextStorage {
+
+    /**
+     * Push.
+     */
+    @Nullable
+    <T extends RequestContext> T push(RequestContext toPush);
+
+    /**
+     * Pop.
+     */
+    void pop(@Nullable RequestContext toRestore);
```

Comment:
Think this is fine as is but for reference I notice grpc accepts a reference to the current context as well, maybe it helps catch bugs.

https://grpc.github.io/grpc-java/javadoc/io/grpc/Context.Storage.html#detach-io.grpc.Context-io.grpc.Context-

Reply:
Ah yes, I didn't add it because we recommend using `push()` with `try-with-resources`, but there's no harm to add the current context. Will add it, thanks!